src/app/forms/components/defects/defects.component.ts

```diff
@@ -1,25 +1,30 @@
 import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { AbstractControl } from '@angular/forms';
 import { CustomFormArray, CustomFormGroup } from '@forms/services/dynamic-form.types';
-import { Defects } from '@models/defects';
 
 @Component({
-  selector: 'app-defects',
-  templateUrl: './defects.component.html',
+  selector: 'app-defects[form]',
+  templateUrl: './defects.component.html'
 })
 export class DefectsComponent {
   @Input() isEditing = false;
-  @Input() defects?: Defects;
+  @Input() form!: CustomFormGroup;
 
   @Output() formsChange = new EventEmitter<(CustomFormGroup | CustomFormArray)[]>();
 
-  forms: Array<CustomFormGroup | CustomFormArray>;
+  get defectsForm() {
+    return this.form.get(['testTypes', '0', 'defects']) as CustomFormArray;
+  }
+
+  getDefectForm(i: number) {
+    return this.defectsForm.controls[i] as CustomFormGroup;
+  }
 
-  constructor() {
-    this.forms = new Array(this.defects?.length).fill(null);
+  trackDefects(index: number, defect: AbstractControl): number {
```

Comment:
Nitpick - this is called `trackByFn` in other components...

```suggestion
  trackByFn(index: number, defect: AbstractControl): number {
```